api/handler/trigger.go

```diff
@@ -39,19 +39,62 @@ func updateTrigger(writer http.ResponseWriter, request *http.Request) {
 		return
 	}
 
+	var problems []dto.TreeOfProblems
+	if isNeedValidate(request) {
+		problems = validateTargets(request, trigger)
+		if problems != nil && dto.HaveTreesError(problems) {
+			writeErrorSaveResponse(writer, request, problems)
+			return
+		}
+	}
+
 	timeSeriesNames := middleware.GetTimeSeriesNames(request)
 	response, err := controller.UpdateTrigger(database, &trigger.TriggerModel, triggerID, timeSeriesNames)
 	if err != nil {
 		render.Render(writer, request, err) //nolint
 		return
 	}
 
+	if problems != nil {
+		response.Check.Targets = problems
+	}
+
 	if err := render.Render(writer, request, response); err != nil {
 		render.Render(writer, request, api.ErrorRender(err)) //nolint
 		return
 	}
 }
 
+func isNeedValidate(request *http.Request) bool {
```

Comment:
Maybe we should rename function to be a correct english phrase

```suggestion
func needsValidation(request *http.Request) bool {
```

Reply:
Following english language rules makes code too perfect, I think it is redundant. There are well naming conventions to use `is, has, can`. Also I think `needValidate` is enough, but `is` prefix is well to stress that it is predicate

Reply:
I think, that predicates should look good, when they are put in condition statement like `if needsValidation(request) { ... }`.

The return type of bool already stresses that this function is a predicate. We already have functions with similar semantics that don't follow this convention (e.g. [this](https://github.com/moira-alert/moira/blob/759d70687dc7ed25768b202a4af177c1744c67cd/checker/worker/metrics.go#L27))

Reply:
> I think, that predicates should look good, when they are put in condition statement

I agree with this approach, then `if needValidate(request)` is the best variant, it is easy to read. In your example `needHandlePattern` we have `need`, not `needs`, there is no `needs` in Moira

Reply:
Ok, then let's have `needValidate`

Reply:
Well, I agree with `needValidate`. @kissken do you agree?

Reply:
yep

Reply:
Done https://github.com/moira-alert/moira/pull/817/commits/7c68d2f125f4275b5d23fc8573896cbc67fc3f9c